src/validation.cpp

```diff
@@ -761,7 +761,14 @@ static bool AcceptToMemoryPoolWorker(const CChainParams& chainparams, CTxMemPool
         size_t nLimitDescendantSize = gArgs.GetArg("-limitdescendantsize", DEFAULT_DESCENDANT_SIZE_LIMIT)*1000;
         std::string errString;
         if (!pool.CalculateMemPoolAncestors(entry, setAncestors, nLimitAncestors, nLimitAncestorSize, nLimitDescendants, nLimitDescendantSize, errString)) {
-            return state.DoS(0, false, REJECT_NONSTANDARD, "too-long-mempool-chain", false, errString);
+            setAncestors.clear();
+            // If the new transaction is relatively small (up to 40k weight)
+            // and has at most one ancestor (ie ancestor limit of 2, including
+            // the new transaction), allow it if its parent has exactly the
+            // descendant limit descendants.
+            if (nSize <= 10000 && !pool.CalculateMemPoolAncestors(entry, setAncestors, 2, 1000000, nLimitDescendants + 1, nLimitDescendantSize + 10000, errString)) {
```

Comment:
I'm still chewing on the comment above, but doesn't this line let transactions >40k weight through no matter what?

Reply:
Lol, oops, one of these days I'll learn to code.

Reply:
did the tests work? ;D

Reply:
@TheBlueMatt I have a spill-over question from the PR review club which covered this PR last week. I am unsure of why the LimitAncestorSize for the carve-out tx has been increased to the block-size. 

If the parent of the carve-out tx were close to the LimitAncestorSize, this would create space for the carve-out tx. Preceding (low-fee-rate) descendent chains of the (carve-out) parent would have to adhere to LimitAncestorSize I believe. 

However, what if the first parent descendent chain was from an actor intending to raise package fee-rate? This (well-intentioned) descendent chain would be burdened by LimitAncestorSize, in contrast to the subsequently broadcasted carve-out transaction, right? In an extreme case, the first descendent chain may even have less than 40k weight to work with?

How should one best reason about this? Thanks.

Reply:
Indeed, we don't really care about the size of the ancestor transaction here (its limited to the standard single-tx size limit, or 100k today), so I just set it to 1M, we already check that there is only, at max, one unconfirmed parent.